File: unity_app/Assets/Scripts/Data/VenueData.cs
using System;
using UnityEngine;
using EventPlannerSim.Core;

namespace EventPlannerSim.Data
{
    /// <summary>
    /// ScriptableObject defining a venue's attributes and capabilities.
    /// Requirements: R9.1
    /// </summary>
    [CreateAssetMenu(fileName = "Venue", menuName = "EventPlanner/Venue")]
    public class VenueData : ScriptableObject
    {
        [Header("Identity")]
        public string id;
        public string venueName;
        public VenueType venueType;
        public VendorTier tier;

        [Header("Capacity")]
        public int capacityMin;
        public int capacityMax;
        public int capacityComfortable;

        [Header("Pricing")]
        public float basePrice;
        public float pricePerGuest;

        [Header("Attributes")]
        public bool isIndoor;
        public bool hasOutdoorSpace;
        public bool weatherDependent;
        
        [Range(1f, 5f)]
        public float ambianceRating;

        [Header("Unlock Requirements")]
        public MapZone zone;
        public BusinessStage requiredStage;
    }
}

File: original-source/Models/Data/VenueData.swift
import Foundation

struct VenueData: Codable, Equatable, Identifiable {
    var id: String
    var venueName: String
    var venueType: VenueType
    var tier: VendorTier
    var capacityMin: Int
    var capacityMax: Int
    var capacityComfortable: Int
    var basePrice: Double
    var pricePerGuest: Double
    var isIndoor: Bool
    var hasOutdoorSpace: Bool
    var weatherDependent: Bool
    var ambianceRating: Double
    var zone: MapZone
    var requiredStage: BusinessStage
}
